lib/travis/yml/configs/travis/repo.rb

```diff
@@ -7,21 +7,35 @@ module Yml
     module Configs
       module Travis
         class Repo < Struct.new(:slug)
-          include Errors
+          include Errors, Helper::Obj
+
+          def authorize(user_token)
+            unauthenticated(slug) unless user_token
+            get(path, auth: { token: user_token })
+          end
 
           def fetch
-            resp = client.get("repo/#{slug.to_s.sub('/', '%2F')}?representation=internal")
-            map(Oj.load(resp.body))
-          rescue Error => e
-            api_error('Travis CI', :repo, slug, e)
+            get(path, auth: internal_auth, representation: :internal)
           end
 
           private
 
+            def path
+              "repo/#{slug.to_s.sub('/', '%2F')}"
```

Comment:
Is there a more explicit way to construct `OWNER%2FREPO`?

Reply:
done, now using `CGI.escape`. thank you!